server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1589,6 +1509,120 @@ static String resolveExpression(String expression, LongSupplier getTime) {
         }
     }
 
+    public static final class ExplicitResourceNameFilter {
+
+        private ExplicitResourceNameFilter() {
+            // Utility class
+        }
+
+        public static List<String> filter(Context context, List<String> expressions) {
+            List<String> result = new ArrayList<>(expressions.size());
+            for (ExpressionList.Expression expression : new ExpressionList(context, expressions)) {
+                validateAliasOrIndex(expression);
+                if (expression.isWildcard() || expression.isExclusion() || ensureAliasOrIndexExists(context, expression.get())) {
+                    result.add(expression.expression());
+                }
+            }
+            return result;
+        }
+
+        @Nullable
+        private static boolean ensureAliasOrIndexExists(Context context, String expression) {
```

Comment:
nit: can add some javadoc with the expectation that if it it exists will return true; else it will either throw (when ignoreUnavalbe is false) or return true when (ignoreUnvailable is true)

Reply:
Pushed [1750624](https://github.com/elastic/elasticsearch/pull/92518/commits/17506245f56512bc79f62276a053e09602e363c2)